Automattic/jetpack
# Packages: Move autoloader tests to the package

Previously the autoloader tests weren't ran automatically, because they were living in `tests/packages`, and we removed that `packages` test suite when we moved the logo tests to the package.

This PR moves the autoloader tests to the package, where they can be ran independently and as part of the builds we run on CI.

#### Changes proposed in this Pull Request:
* Move the autoloader tests to the package
* Add automated testing infrastructure to the package.
* No longer rely on the WP unit test framework.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
Part of the Jetpack DNA project - p1HpG7-70O-p2

#### Testing instructions:
* Checkout this branch.
* Navigate to the package: `cd packages/autoloader`
* Install and run tests: `composer phpunit`
* Verify tests pass.

#### Proposed changelog entry for your changes:
* Move autoloader tests to the package
